src/vtl.ts

```diff
@@ -186,6 +195,39 @@ export class VTL {
         }
       }
       case "BinaryExpr":
+        if (node.op === "in") {
+          /**
+           * rewrite `in` to a conditional statement to support both arrays and maps
+           * var v = left in right;
+           *
+           * var v = right.class.name === "ArrayList" ?
+           *    right.length >= left :
+           *    right.containsKey(left);
+           */
+          const condition = new ConditionExpr(
+            new BinaryExpr(
+              new PropAccessExpr(
+                new PropAccessExpr(node.right, "class"),
+                "name"
+              ),
+              "==",
+              new StringLiteralExpr("ArrayList")
+            ),
+            new BinaryExpr(
+              new PropAccessExpr(node.right, "length"),
+              ">=",
+              node.left
+            ),
+            new CallExpr(new PropAccessExpr(node.right, "containsKey"), [
+              new Argument(node.left),
+            ])
+          );
+          condition.setParent(node);
```

Comment:
Should we do this re-write prior to interpreting with a `visitEachChild` transformer?

Reply:
1. Wouldn't the logic be essentially the same without the extra pass?
2. Could I do that when I refactor app sync later?

Reply:
1. Yeah but you wouldn't have to `setParent` yourself.
2. sure, no biggie, but it would be pretty easy to do
```ts
constructor() {
  const transformedAst = visitEachChild(functionDecl, function normalizeAST(node) {
    if (isBinaryOp(node) && .. ) {
       // same code you already have, minus setParent
    }
  });
}
```